### PR TITLE
Dev/13

### PR DESCRIPTION
**Pull request by PaMeirelles**
Used the process described in [this ](https://sol.sbc.org.br/index.php/eniac/article/view/18427/18260)article as basis.

### CODE REVIEW

**Comment by PaMeirelles** [comment]
@lucasximenes I fixed the issues with the Stage Disturb. Should be working fine now.

**Comment by PaMeirelles** [comment]
Would also like to add that this version of the code was able to find the optimal solution for a 76 node instance in less than a minute (although not reliably), which is not huge but a cute little achievement nonetheless!
image
image

File: src/StageDisturb.jl [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        while room_to_improve
+            room_to_improve = false
+            shuffle!(stageDisturb.neighbourSearch)
+            for ns in stageDisturb.neighbourSearch
+                if bestImprovement!(ns)
+                    room_to_improve = true
+                    updateCost!(ns.solution, ns.data.weights)
+                    break
+                else 
+                    updateCost!(ns.solution, ns.data.weights)
```
Comment (lucasximenes) [inline src/StageDisturb.jl]: why update cost if neighborhood failed to improve?

File: src/StageDisturb.jl [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                end
+            end
+        end
+
+        currSol = stageDisturb.neighbourSearch[1].solution
```
Comment (lucasximenes) [inline src/StageDisturb.jl:38]: why do you always set currSol as the solution from the first neighborhood? if neighbourSearch is shuffled somewhere, then I agree with this, but I didn't see it

File: src/StageDisturb.jl (hunk 1 above)
Comment (PaMeirelles) [reply src/StageDisturb.jl]: You are right, we don't need it. I will remove the else clause.

File: src/StageDisturb.jl (hunk 2 above)
Comment (PaMeirelles) [reply src/StageDisturb.jl:38]: All neighbourSearchs point to the same array, so it is indifferent which one I use to update currSol

Comment (PaMeirelles) [reply src/StageDisturb.jl:38]: @lucasximenes I checked the code again and you are right, I was deepcopying to every NS when I meant to create a copy and pass the reference. I've pushed the new changes.

**Pull request merged**